int32 value
---
int32 oldValue
int32 newValue